[Bug]  fails when generating ci-i386 image: xgboost requires at least CMake 3.14

The nightly job that rebuilds Docker images is failing to generate `ci-i386` due to an error when installing xgboost as part of https://github.com/apache/tvm/blob/main/docker/install/ubuntu_install_redis.sh.

The error is:
```
 copying xgboost/LICENSE -> build/temp.linux-x86_64-3.6/xgboost
  INFO:XGBoost build_ext:Building from source. /tmp/pip-install-3wzms0ia/lib/libxgboost.so
  INFO:XGBoost build_ext:Run CMake command: ['cmake', 'xgboost', '-GUnix Makefiles', '-DUSE_OPENMP=1', '-DUSE_CUDA=0', '-DUSE_NCCL=0', '-DBUILD_WITH_SHARED_NCCL=0', '-DHIDE_CXX_SYMBOLS=1', '-DUSE_HDFS=0', '-DUSE_AZURE=0', '-DUSE_S3=0', '-DPLUGIN_DENSE_PARSER=0']
  CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
    CMake 3.14 or higher is required.  You are running version 3.13.5
  
  
  -- Configuring incomplete, errors occurred!
  WARNING:XGBoost build_ext:Disabling OpenMP support.
  INFO:XGBoost build_ext:Run CMake command: ['cmake', 'xgboost', '-GUnix Makefiles', '-DUSE_OPENMP=0', '-DUSE_CUDA=0', '-DUSE_NCCL=0', '-DBUILD_WITH_SHARED_NCCL=0', '-DHIDE_CXX_SYMBOLS=1', '-DUSE_HDFS=0', '-DUSE_AZURE=0', '-DUSE_S3=0', '-DPLUGIN_DENSE_PARSER=0']
  CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
    CMake 3.14 or higher is required.  You are running version 3.13.5
  
  
  -- Configuring incomplete, errors occurred!
  Traceback (most recent call last):
    File "/tmp/pip-install-3wzms0ia/xgboost/setup.py", line 172, in build_cmake_extension
      self.build(src_dir, build_dir, gen, build_tool, use_omp=1)
    File "/tmp/pip-install-3wzms0ia/xgboost/setup.py", line 115, in build
      subprocess.check_call(cmake_cmd, cwd=build_dir)
    File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['cmake', 'xgboost', '-GUnix Makefiles', '-DUSE_OPENMP=1', '-DUSE_CUDA=0', '-DUSE_NCCL=0', '-DBUILD_WITH_SHARED_NCCL=0', '-DHIDE_CXX_SYMBOLS=1', '-DUSE_HDFS=0', '-DUSE_AZURE=0', '-DUSE_S3=0', '-DPLUGIN_DENSE_PARSER=0']' returned non-zero exit status 1.
  
```

Log: https://ci.tlcpack.ai/blue/rest/organizations/jenkins/pipelines/docker-images-ci/pipelines/daily-docker-image-rebuild/runs/96/nodes/58/steps/220/log/?start=0

@tqchen who can we ask to help fixing this issue?

--- a/docker/install/ubuntu_install_cmake_source.sh
+++ b/docker/install/ubuntu_install_cmake_source.sh
@@ -20,8 +20,8 @@ set -e
 set -u
 set -o pipefail
 
-v=3.13
-version=3.13.5
+v=3.14
+version=3.14.7
 wget https://cmake.org/files/v${v}/cmake-${version}.tar.gz
 tar xvf cmake-${version}.tar.gz
 cd cmake-${version}

